modules/programs/zsh.nix

```diff
@@ -345,9 +345,14 @@ in
         ++ optional cfg.enableCompletion nix-zsh-completions
         ++ optional cfg.oh-my-zsh.enable oh-my-zsh;
 
-      home.file."${relToDotDir ".zshrc"}".text = ''
+      home.file."${relToDotDir ".zshenv"}".text = ''
         typeset -U path cdpath fpath manpath
 
+        # Set PATH for both interactive and non-interactive shell
+        path+=($HOME/.nix-profile/bin /etc/profiles/per-user/$USER/bin /nix/var/nix/profiles/default/bin /run/current-system/sw/bin)
```

Comment:
Will this interact with the `$PATH` variable any? In particular I'm curious about if there is a risk for problems on non-NixOS systems or inside a Nix shell where `PATH` has been enriched. Like

```sh
#! /usr/bin/env nix-shell
#! nix-shell -i zsh -p hello
hello
```

Reply:
Hmm, this is a very good point... It works fine with your example, but it modifies behavior when `--pure` flag is passed by appending all the directories above:

```zsh
#! /usr/bin/env nix-shell
#! nix-shell -i zsh --pure -p zsh

echo $PATH
```

I'm wondering how the problem described in #915 is fixed for bash?

Reply:
Sorry for the long delay! When Bash is invoked as a remote command it will source `~/.bashrc` (and `/etc/bashrc`) in non-interactive mode, see [Invoked by remote shell daemon](https://www.gnu.org/software/bash/manual/html_node/Bash-Startup-Files.html#Invoked-by-remote-shell-daemon). Since the NixOS `/etc/bashrc` contains

```bash
# If the profile was not loaded in a parent process, source
# it.  But otherwise don't do it because we don't want to
# clobber overridden values of $PATH, etc.
if [ -z "$__ETC_PROFILE_DONE" ]; then
    . /etc/profile
fi
```

it will, among other things, set up the `PATH` variable. So as far as I can tell this will set up the Bash environment to be very similar to a non-interactive login shell with the notable exception being that it wouldn't source `~/.profile` and hence not source `hm-session-vars.sh`. This matches what I see, running `ssh walnut env` where my user is managed by HM prints variables set by `/etc/profile` but not ones set by `hm-session-vars.sh`.

Finally, `/etc/profile` has
```bash
# Only execute this file once per shell.
if [ -n "$__ETC_PROFILE_SOURCED" ]; then return; fi
__ETC_PROFILE_SOURCED=1
```
at the top, which I guess is the underlying reason why we don't get duplicate `PATH` entries in Bash.

The HM Bash module actually never touches the `PATH` variable and instead completely relies on the system configuration setting it up correctly. I wonder if perhaps the same should be for zsh?